File: MVVMDemo/app/src/main/java/com/cyq/mvvmdemo/MainActivity.java
package com.cyq.mvvmdemo;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.cyq.mvvmdemo.databinding.ActivityMainBinding;
import com.cyq.mvvmdemo.vm.LoginViewModel;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        new LoginViewModel(binding);
    }
}

File: NestedScroll/app/src/main/java/com/chenyangqi/nested/scroll/MainActivity.kt
package com.chenyangqi.nested.scroll

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Parcel
import android.os.Parcelable
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {
    private lateinit var mRecyclerView: RecyclerView
    private lateinit var mAdapter: MyAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val mDatas: MutableList<String> = ArrayList()
        for (i in 1..100) {
            mDatas.add("item = $i")
        }

        mRecyclerView = findViewById<RecyclerView>(R.id.mRecyclerView)
        mAdapter = MyAdapter(mDatas)
        mRecyclerView.adapter = mAdapter
        mRecyclerView.layoutManager = LinearLayoutManager(this)
    }

    inner class MyAdapter(val data: MutableList<String>) : RecyclerView.Adapter<MyViewHolder>() {
        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {

        }

        override fun onBindViewHolder(holder: MyViewHolder, position: Int) {

        }

        override fun getItemCount(): Int {

        }

    }

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {


    }

}